internal/dag/cache.go

```diff
@@ -552,13 +552,13 @@ func (kc *KubernetesCache) LookupSecret(name types.NamespacedName, validate func
 	return s, nil
 }
 
-func (kc *KubernetesCache) LookupUpstreamValidation(uv *contour_api_v1.UpstreamValidation, namespace string) (*PeerValidationContext, error) {
+func (kc *KubernetesCache) LookupUpstreamValidation(uv *contour_api_v1.UpstreamValidation, caCertificate types.NamespacedName) (*PeerValidationContext, error) {
 	if uv == nil {
 		// no upstream validation requested, nothing to do
 		return nil, nil
 	}
 
-	secretName := types.NamespacedName{Name: uv.CACertificate, Namespace: namespace}
+	secretName := types.NamespacedName{Name: caCertificate.Name, Namespace: caCertificate.Namespace}
```

Comment:
Looks like you can drop the `secretName` var altogether and just reference `caCertificate` directly where needed

Reply:
Makes sense, thanks!